students/tri_nguyen/lesson05/mailroom.py

```diff
@@ -0,0 +1,201 @@
+    4. Quit
+
+    ''')
+
+
+def display_donor_list():
+    ''' this function displays list of donors
+        from donor_dict '''
+
+    donor_name = enumerate([donor for donor in donor_dict])
+    for idx, name in donor_name:
+        print(idx + 1, '-', name)
+
+
+def convert_to_float(num):
+    ''' convert numeric values from string to float '''
+
+    try:
+        num = float(num)
+    except ValueError:
+        print('value entered is not a number!!')
+    else:
+        return num
+    
+
+
+def send_thanks():
+    ''' this function composes an email thanking
+        the donor for their generous donation '''
+
+    while True:
+
+        full_name = input('Enter a Full Name (or type "list" to get the list of donors): ').title()
+
+        if full_name.lower() == 'list':
+            display_donor_list()
+            continue
+
+        elif donor_dict.get(full_name):
+            donation_amount = convert_to_float(input('Enter donation amount: '))
+            # Update values in list
+            donor_dict[full_name][0] = donor_dict[full_name][0] + donation_amount
+            donor_dict[full_name][1] = donor_dict[full_name][1] + 1
+            donor_dict[full_name][2] = donor_dict[full_name][0] / donor_dict[full_name][1]
+
+        else:
+            donation_amount = convert_to_float(input('Enter donation amount: '))
+            donor_dict[full_name] = [donation_amount, 1, donation_amount]
+
+        choice = input('Do you want to quit and send email?(y/n) ')
+
+        if choice == 'y':
+            email_template = '''
+                Dear {0},
+
+                Thank you for giving us ${1:,.2f}. Your money will be put to good use.
+                Please come back and donate more.
+
+                Best Regards,
+
+                John Doe
+            '''
+            print(email_template.format(full_name, donation_amount))
+
+            break
+        else:
+            continue
+
+
+def create_report():
+    ''' this function generates a report of donors '''
+
+    template = '{0:<26}|{1:^13}|{2:^11}|{3:^14}'
+    value_template = '{0:<26} ${1:12.2f} {2:11} ${3:13.2f}'
+
+    formatted_columns = template.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift')
+
+    print(formatted_columns)
+
+    print('{}'.format('-' * len(formatted_columns)))
+
+    print('Report of Donors is below:\n')
+
+    donor_list = list(donor_dict.items())
+
+    donor_list.sort(key=itemgetter(1), reverse=True)
+
+    for item in donor_list:
+        print(value_template.format(item[0], item[1][0], item[1][1], item[1][2]))
+    print()
+
+
+def format_filename(name):
+    ''' format name to create a filename that looks like
+        William_Gates_III.txt '''
+
+    if ',' in name:
+        name = name.replace(',', '')
+        name = name.replace(' ', '_')
+        name = name + '.txt'
+        return name
+    else:
+        name = name.replace(' ', '_')
+        name = name + '.txt'
+        return name
```

Comment:
I think you can just have all of this in one block:
```
def format_filename(name):
    name = name.replace(',', '')
    name = name.replace(' ', '_')
    return name + '.txt'
```
replace will not do anything if `,` is not in name